### PR TITLE
56297 omb component migration

### LINKED ISSUE #56297: Migrate OMBInfo component on 526 Introduction Page
## Issue Description
Migrate the[ Introduction Page ](https://github.com/department-of-veterans-affairs/vets-website/blob/8e3923d906a3427570d234d6d329f2fd6c2f20f8/src/applications/disability-benefits/all-claims/components/IntroductionPage.jsx#L254)to use the [web components version of OMB Info](https://design.va.gov/storybook/?path=/docs/components-va-omb-info--default) instead of deprecated react component

Use the current values for now. Content will be updated with #56510

https://www.va.gov/disability/file-disability-claim-form-21-526ez/

When clicking on the Privacy Act Statement link, modal opens

## Tasks
- [x] Figure out what the component info needs to be.

## Questions
- How would the accessibility tests be conducted?

## Front end tasks
- Ensure the page is styled according to design spec. 
- Ensure the back end data from API is displayed in local and review instance
- Ensure unit tests are available. 
- Ensure your code changes are covered by E2E tests
- Run axe checks using the Chrome or Firefox browser plugin
- Test for color contrast and color blindness issues
- Zoom layouts to 400% at 1280px width
- Test with 1 or 2 screen readers
- Navigate using the keyboard only

## Acceptance Criteria
- [ ] Designer comments on this issue to approve page styling 
- [ ] Unit tests pass
- [ ] E2E tests cover current code and regression
- [ ] End-to-end tests show 0 violations.
- [ ] The data returned matches API response (for given user or scenario)
- [ ] All axe checks pass
- [ ] All color contrast checks pass
- [ ] All zoom testing passes
- [ ] All keyboard checks pass 
- [ ] All screenreader checks pass

## Appendix
<details> 

 <summary>Ensure your code changes are covered by E2E tests (expand)</summary>
 
 - Add E2E tests if none exist for this addition/change. 
 
 - Update existing E2E tests if this code will change functionality. 
 
 - Include axe checks, including hidden content
 
</details>

<details> 

 <summary> Run axe checks using the Chrome or Firefox browser plugin (expand)</summary>
 
- Ensure no heading levels are skipped.

- Ensure all buttons and labeled inputs use semantic HTML elements.

- Ensure all buttons, labeled elements and images are identified using HTML semantic markers or ARIA roles.

- Ensure form fields have clearly defined boundaries or outlines.

- Ensure form fields do not use placeholder text. 

- Ensure form fields have highly visible and specific error states. 

</details>

<details> 

 <summary> Test for color contrast and color blindness issues (expand) </summary>
 - All text has appropriate contrast. 

</details>

<details> 

 <summary> Zoom layouts to 400% at 1280px width (expand)</summary> 
 - Ensure readability and usability are supported when zoomed up to 400% at 1280px browser width
 - Ensure no content gets focused offscreen or is hidden from view.
 
</details>

<details> 

 <summary> Test with 1 or 2 screen readers (expand)</summary> 
 
 - Ensure the page includes a skip navigation link. 
 
 - Ensure all links are properly descriptive. 
 
 - Ensure screen reader users can hear the text equivalent for each image conveying information. 

 - Ensure screen reader users can hear the text equivalent for each image button. 

 - Ensure screen reader users can hear labels and instructions for inputs. 
 
 - Ensure purely decorative images are not announced by the screenreader. 
 
</details>

<details> 

 <summary>Navigate using the keyboard only (expand)</summary> 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be reached with the tab key in a logical order. 
 
 - Ensure all links (navigation, text and/or image), form controls and page functions can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure all interactive elements can be reached with the tab key in a logical order 
 
 - Ensure all interactive elements can be triggered with the spacebar, enter key, or arrow keys.
 
 - Ensure focus is always visible and appears in logical order.
 
 - Ensure each interactive element has visible focus state which appears in logical order.
 

</details>

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

### PR DESCRIPTION
**Pull request by christinec-fftc**
## Summary

- Move from deprecated react component to web components version for OMB Info
- This component is displayed near the bottom of the Introduction page

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
[department-of-veterans-affairs/va.gov-team#56297](https://github.com/department-of-veterans-affairs/va.gov-team/issues/56297)

## Testing done
- _Describe what the old behavior was prior to the change_
The page was previously using the react version of the OMB info component which is now deprecated
- _Describe the steps required to verify your changes are working as expected_
Visit the intro page and perform all the usual front end tests. Also click on the privacy link to make sure the modal is working as expected
- _Describe the tests completed and the results_
Unit, e2e and manual tests were successful

## Screenshots
image
image

## What areas of the site does it impact?
form 526ez

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [x] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/constants.js [hunk 1]
```diff
@@ -337,3 +337,5 @@ export const CHAR_LIMITS = [
 
 // migration max string length
 export const MAX_HOUSING_STRING_LENGTH = 500;
+
+export const OMB_CONTROL = '2900-0747';
```
Comment (mchae-nava) [inline src/applications/disability-benefits/all-claims/constants.js:341]: Just curious, what is the OMB number?

**Review by mchae-nava — APPROVED** [review]
LGTM!

File: src/applications/disability-benefits/all-claims/constants.js (hunk 1 above)
Comment (christinec-fftc) [reply src/applications/disability-benefits/all-claims/constants.js:341]: It's a unique number assigned to forms. You can see it on the top right corner of 1st page of the paper/pdf version of 526. Also definition...

> OMB Numbers are form and information collection numbers assigned by The Office of Management and Budget. The Office of Management and Budget is the largest office within the Executive Office of the President of the United States. The OMB generates these Control Numbers under the Paperwork Reduction Act of 1980.

Comment (mchae-nava) [reply src/applications/disability-benefits/all-claims/constants.js:341]: Ahh gotcha! Thanks!

**Review by aurora-a-k-a-lightning — APPROVED** [review]

**Review by sethdarragile6 — APPROVED** [review]

**Review by adamwhitlock1 — APPROVED** [review]
this lgtm. The failing unit tests I think have been updated in main so if you update this branch I think they should pass CI/CD then.

**Pull request merged**